### PR TITLE
[MAINTENANCE] Use `set` instead of `create` in DatasourceStore

### PR DESCRIPTION
**Pull request by anthonyburdi**
Changes proposed in this pull request:
- Refactor to use `set` instead of `create` in `DatasourceStore`
- Refactor `test_datasource_crud.py` to use parametric tests for each data context type
- Partial refactor of private datasource creation methods to not require a name in addition to the name being in the `DatasourceConfig`
- Note: schema roundtrip to deal with id vs id_ is being handled in a separate PR
- Note: This is a partial implementation of code from https://github.com/great-expectations/great_expectations/pull/5664

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 1]
```diff
@@ -1594,12 +1600,47 @@ def _instantiate_datasource_from_config(
...
+        # Round trip through schema validation and config creation to ensure "id_" is present
+        #
+        # Chetan - 20220804 - This logic is utilized with other id-enabled objects and should
+        # be refactored to into the config/schema. Also, downstream methods should be refactored
+        # to accept the config object (as opposed to a dict).
         if isinstance(config, DatasourceConfig):
-            config = datasourceConfigSchema.dump(config)
+            config: dict = datasourceConfigSchema.dump(config)
+            validated_config: DatasourceConfig = datasourceConfigSchema.load(config)
+            config: dict = validated_config.to_json_dict()
```
Comment (Kilo59) [inline great_expectations/data_context/data_context/abstract_data_context.py]: Can now use `DatasourceConfig._dict_round_trip()` here.

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 2]
```diff
@@ -1632,25 +1676,37 @@ def _instantiate_datasource_from_config_and_update_project_config(
...
+        # Round trip through schema validation and config creation to ensure "id_" is present
+        #
+        # Chetan - 20220804 - This logic is utilized with other id-enabled objects and should
+        # be refactored to into the config/schema. Also, downstream methods should be refactored
+        # to accept the config object (as opposed to a dict).
+        substituted_config: DatasourceConfig = datasourceConfigSchema.load(
+            substituted_config_dict
+        )
+        schema_validated_substituted_config_dict: dict = (
+            substituted_config.to_json_dict()
```
Comment (Kilo59) [inline great_expectations/data_context/data_context/abstract_data_context.py]: `DatasourceConfig._dict_round_trip()`

File: great_expectations/data_context/store/ge_cloud_store_backend.py [hunk 3]
```diff
@@ -333,22 +359,34 @@ def _set(  # type: ignore[override]
...
+    def _retrieve_id_from_response(self, response: requests.Response) -> str:
+        """Pull out the id from the response object json
+
+        Args:
+            response: response to retrieve id from
+
+        Returns:
+            string of id
+        """
+        return response.json()["data"]["id"]
```
Comment (Kilo59) [inline great_expectations/data_context/store/ge_cloud_store_backend.py:447]: I think we should take this opportunity to create a custom object that wraps the response and exposes an `id` attribute or `@property`.

I'd be happy to take this on.

**Comment by anthonyburdi** [comment]
Was incorporated in https://github.com/great-expectations/great_expectations/pull/5941, https://github.com/great-expectations/great_expectations/pull/5901, https://github.com/great-expectations/great_expectations/pull/5813 and others in Aug 2022

**Pull request closed without merging**